torch/_inductor/compile_worker/subproc_pool.py

```diff
@@ -157,31 +187,34 @@ def submit(
         return future
 
     def _read_thread(self) -> None:
-        try:
-            while True:
-                job_id, data = _recv_msg(self.read_pipe)
-                if job_id < 0:
-                    if self.running:
-                        log.warning("SubprocPool unclean exit")
-                    self.read_pipe.close()
+        while True:
+            job_id, data = _recv_msg(self.read_pipe)
```

Comment:
Are you sure we no longer need a try around the recv? I'd assume that could raise if the pipe on the other end closes, for example

Reply:
I looked at it closer. I think we're both right - with the except in the place it was before then there was a problem with exceptions being caught and ignored. In the next version I've added a try around the recv which then notifies the future about the exception.